### PR TITLE
Filter/pagination implementation for getWorkflows API

### PR DESCRIPTION
**Pull request by kofi-clever**
Adds support for sorting/filtering/pagination to the `getWorkflows` API.
Uses a new secondary index on the concatenated workflow definition name and status: https://github.com/Clever/infra/pull/641

- [N/A] Update swagger.yml version
- [x] Run "make generate"
- [N/A] After merging, add a github tag to your merge commit

### CODE REVIEW

File: store/dynamodb/dynamodb_store.go [hunk 1]
```diff
@@ -540,37 +552,61 @@ func (d DynamoDB) GetWorkflowByID(id string) (resources.Workflow, error) {
...
+		return workflows, nextPageToken, err
 	}
 
 	for _, item := range res.Items {
 		workflow, err := DecodeWorkflow(item)
 		if err != nil {
-			return workflows, err
+			return workflows, nextPageToken, err
 		}
 		if err := d.populateWorkflow(&workflow); err != nil {
```
Comment (mohit) [inline store/dynamodb/dynamodb_store.go]: not your change but this should be `populateWorkflowDefinition(&workflow)` also we should add a TODO or comment that we could use `ddb.BatchGet` for this eventually.

File: store/dynamodb/page_key.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package dynamodb
```
Comment (mohit) [inline store/dynamodb/page_key.go]: FYI: I'm not familiar with paging much but looks good from my perspective

File: store/store.go [hunk 3]
```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/Clever/workflow-manager/resources"
 )
 
-// WorkflowStore defines the interface for persistence of Workflow defintions
+// Store defines the interface for persistence of Workflow definitions.
```
Comment (mohit) [inline store/store.go]: s/Workflow definitions/Workflow Manager resources/

File: store/tests/tests.go [hunk 4]
```diff
@@ -199,29 +200,113 @@ func GetWorkflowByID(s store.Store, t *testing.T) func(t *testing.T) {
...
+	}
+}
 
-		workflows, err := s.GetWorkflows(wf2.Name())
-		require.Nil(t, err)
-		require.Equal(t, len(workflows), len(workflow2IDs))
-		var gotWorkflow2IDs []string
-		for _, j := range workflows {
-			gotWorkflow2IDs = append(gotWorkflow2IDs, j.ID)
+func GetWorkflowsPagination(s store.Store, t *testing.T) func(t *testing.T) {
```
Comment (mohit) [inline store/tests/tests.go]: should we test for some pagination error cases here?

File: store/dynamodb/dynamodb_store.go (hunk 1 above)
Comment (kofi-clever) [reply store/dynamodb/dynamodb_store.go]: agreed - https://github.com/Clever/workflow-manager/pull/50/commits/e840927ec091bd7e5ed5b581c7d994345812a1bf

File: store/store.go (hunk 3 above)
Comment (kofi-clever) [reply store/store.go]: oh yeah, good catch - https://github.com/Clever/workflow-manager/pull/50/commits/907f371759bd1c7450ebd587f987ae9ef7516bc9

File: store/tests/tests.go (hunk 4 above)
Comment (kofi-clever) [reply store/tests/tests.go]: +1 - https://github.com/Clever/workflow-manager/pull/50/commits/d22346cf0ea6da0a2fc1b0977f13c695d95c306a

**Pull request merged**